### ISSUE TITLE
BatchNorm1d + fusing into a linear layer

### ISSUE THREAD

**Issue by j2kun** [google/heir#2997]
In https://github.com/google/heir/issues/2923 we have BatchNorm1d as an uncovered operator. In some more detail, a common technique in FHE is to use BatchNorm instead of Dropout for regularization, and then to fuse the BatchNorm into the preceding linear layer.

For example, the torch might look like

```python
nn.Sequential(
    ...
    nn.Linear(in_size, out_size),
    nn.BatchNorm1d(out_size),
    nn.ReLU(),
    ...
)
``` 

The batch norm 1d produces four trained parameters, gamma, beta, mu and sigma, and during inference the layer computes this formula elementwise over its input.

<img width="314" height="99" alt="Image" src="https://github.com/user-attachments/assets/96b353fe-2239-4e8e-a11d-f44ce9cfca6f" />

What we want to do is:

1. Set up a sample torch model with these three layers, and see what HEIR does with it today.
2. Determine the best way to implement the fusion. Probably it will be during the [linalg-canonicalizations](https://github.com/google/heir/blob/main/lib/Transforms/LinalgCanonicalizations/LinalgCanonicalizations.td), but the question is how to identify the op or sequence of ops that BatchNorm1d lowers to.
3. Determine a plan for the remaining use cases of BatchNorm1d:
   1. Implementing the op as plaintext-ciphertext mul/add (this _should_ be close to working already...)
   2. Determining what other ops it potentially can be fused with (e.g., a preceding convolution)

**Comment by AlexanderViand** [comment]
What we've been doing internally is to do the batchnorm fusing pre-HEIR in python/pytorch, which seems a bit easier

EDIT: what HEIR _used_ to do with this is to try and approximate `sqrt` over the default interval of `[-1,1]` which caused a failure down the line in `polynomial.eval`, but now that you recently changed the default interval for square root to `[0,1]` I actually don't know what HEIR does with this

**Comment by j2kun** [comment]
> EDIT: what HEIR _used_ to do with this is to try and approximate `sqrt` over the default interval of `[-1,1]` which caused a failure down the line in `polynomial.eval`, but now that you recently changed the default interval for square root to `[0,1]` I actually don't know what HEIR does with this

A bit confused because the square root is applied to constants, so there should be no approximation involved. I would expect it materializes as a set of plaintext-ciphertext ops and the division becomes a product by the reciprocal.

**Comment by AlexanderViand** [comment]
Is the approximation pass even secretness-aware? I'm pretty sure I've demoed it on vanilla mlir before and it still worked

**Comment by j2kun** [comment]
I started work on basic support for batch norm 1d (without fusion) as a baseline.
